crypto/zk/artifacts/artifacts.go

```diff
@@ -0,0 +1,179 @@
+package artifacts
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+/*
+# Directories structure
+=======================
+Path: circuitname/network/parameters/file.extension
+- Circuit Name: the circuit name. Example: `zkcensusproof`
+- Network: The network name. Example: `main, dev, stage`
+- Circuit Parameters: numbers in decimal separated by underscore. Example: `1024`, `1024_3_500`
+- Files:
+	- circuit.zkey: contains the provingKey & verificationKey (big size, linear with circuit
+	size) [needed by the user]
+	- verificationKey.json: extracted from circuit.zkey (small size, constant size) [needed
+	by the vochain]
+	- witness.wasm: (medium size, linear with circuit size) [needed by the user]
+- Examples:
+	- zkcensusproof/stage/1024/circuit.zkey
+	- zkcensusproof/dev/128/witness.wasm
+
+# Files hash
+=============
+Each file is hashed with sha256, which output is publicly known. The hash
+values of each file are hardcoded in the VochainGenesis.
+
+*/
+
+const (
+	// FilenameWitness defines the name of the file of the WitnessCalculator
+	FilenameWitness string = "witness.wasm"
+	// FilenameZKey defines the name of the file of the circom ZKey
+	FilenameZKey string = "circuit.zkey"
+	// FilenameVK defines the name of the VerificationKey.json
+	FilenameVK string = "verificationKey.json"
```

Comment:
nit: best to leave constants untyped in general, it's easier and more flexible - just `const foo = "value"`